packages/jest-repl/src/cli/repl.ts

```diff
@@ -25,17 +25,25 @@ const evalCommand: repl.REPLEval = (
   let result;
   try {
     if (transformer) {
-      const transformResult = transformer.process(
-        cmd,
-        jestGlobalConfig.replname || 'jest.js',
-        jestProjectConfig,
-      );
-      cmd =
-        typeof transformResult === 'string'
-          ? transformResult
-          : transformResult.code;
+      if (!transformer.process) {
+        throw new TypeError(
+          'Jest: currently only sync transforms are supported; a transform must export a `process function',
+        );
+      }
+      // TODO: support async as well
+      if (transformer.process) {
```

Comment:
you can remove this `if` - if TypeScript says "no" just add a `!` - you have a guard above

Reply:
I was thinking when async support is added, we can just remove line 28 - line 33 without touching line 34. But I can remove this in the subsequent commit 🙂

Reply:
I was thinking when async support is added, we can just remove line 28 - line 33 without touching line 34. But I can remove this in the subsequent commit 🙂